Add Tooltip component and remove EnrichedLabel

joshuatf opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR extracts the Tooltip portion from the EnrichedLabel and creates a separate component for that.  We have multiple use cases for the tooltip that fall outside of labels, so I think it's worthwhile for it to be its own component.

This PR also adds some accessibility keyboard handlers to the tooltip button, though admittedly this does not follow the traditional tooltip pattern and behaves more like a modal (since we can click content inside of the tooltip).

cc @octaedro want to make sure you're on board with this change and I'm not breaking things by removing the `EnrichedLabel`.

<img width="329" alt="Screen Shot 2022-10-10 at 8 57 12 AM" src="https://user-images.githubusercontent.com/10561050/194908164-0493d53e-5972-4657-838b-1dd553129ae5.png">
<img width="385" alt="Screen Shot 2022-10-10 at 8 57 07 AM" src="https://user-images.githubusercontent.com/10561050/194908167-eb9b6217-8e47-4a39-95f5-a52b788e9d83.png">


Closes #34997 .

### How to test the changes in this Pull Request:

1. Run `pnpm --filter=@woocommerce/storybook storybook`
2. Go to Experimental->Tooltip
3. Check that the stories work as expected
4. Attempt to open and close the tooltip via keyboard (enter/escape on focus)
5. Navigate to the new product management experience (Products -> Add new (MVP))
6. Make sure the tooltip next to `Feature this product` continues to work as expected

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

joshuatf committed:
Add experimental tooltip component

joshuatf committed:
Add tooltip stories

joshuatf committed:
Update EnrichedLabel component

joshuatf committed:
Remove EnrichedLabel

joshuatf committed:
Add changelog entries

joelclimbsthings commented:
Hey @joshuatf , I'm unable to `build` this branch due to this error:

```
ERROR in client/products/sections/product-details-section.tsx:16:2
woocommerce/client/admin:turbo:build: TS2305: Module '"@woocommerce/components"' has no exported member 'EnrichedLabel'.
woocommerce/client/admin:turbo:build:     14 | import { cleanForSlug } from '@wordpress/url';
woocommerce/client/admin:turbo:build:     15 | import {
woocommerce/client/admin:turbo:build:   > 16 | 	EnrichedLabel,
woocommerce/client/admin:turbo:build:        | 	^^^^^^^^^^^^^
woocommerce/client/admin:turbo:build:     17 | 	Link,
woocommerce/client/admin:turbo:build:     18 | 	useFormContext,
woocommerce/client/admin:turbo:build:     19 | 	__experimentalTooltip as Tooltip,
```
Is that not something you're seeing?

joshuatf commented:
@joelclimbsthings This was working for me, but probably due to a cached components build since the WCA client and components don't seem to be synced anymore.

Removed that import and gave this another rebase.

joshuatf committed:
Fix up linting issues

joelclimbsthings commented on packages/js/components/src/tooltip/tooltip.tsx:
```diff
@@ -0,0 +1,61 @@
+						}
+						setIsPopoverVisible( true );
+					} }
+					onClick={ () => setIsPopoverVisible( true ) }
```
Should we make this a toggle? I'm not sure if it's just me, but I'd expect the tooltip to close if I click it while open.

joshuatf replied:
Good thinking.  Updated in https://github.com/woocommerce/woocommerce/pull/35024/commits/c55b0626e85a4c9e0a242f08f6c73776f7e85279.

joelclimbsthings commented on packages/js/components/src/tooltip/tooltip.tsx:
```diff
@@ -0,0 +1,61 @@
+				{ isPopoverVisible && (
+					<Popover
+						focusOnMount="container"
+						position="top center"
```
It might be a good idea to make this configurable via the Tooltip props as well. I can see use cases where we'd want to control how the Tooltip positions itself more easily.

joshuatf replied:
Added in https://github.com/woocommerce/woocommerce/pull/35024/commits/c55b0626e85a4c9e0a242f08f6c73776f7e85279.

joelclimbsthings reviewed (changes requested):
Thanks @joshuatf , this looks good overall. I had a couple minor comments, and also noticed some "freak-out" behavior on the tooltip on smaller viewports:

![Peek 2022-10-18 12-40](https://user-images.githubusercontent.com/444632/196528291-a303be84-ee3a-4804-a98e-3ccac79926d1.gif)

joshuatf committed:
Handle PR feedback

joshuatf commented:
Thanks for the review, @joelclimbsthings!  Addressed most of your feedback.

>also noticed some "freak-out" behavior on the tooltip on smaller viewports:

I think this has been an ongoing issue with anchored elements in Gutenberg and something that might need to be handled upstream (or already has but our bundled version of components in Storybook may be dated).

joelclimbsthings reviewed (approved):
Sounds good @joshuatf ; looks good and working well!
